### ISSUE TITLE
NaNs in samples.tsv

### ISSUE THREAD

**Issue by AroArz** [metagenome-atlas/atlas#695]
Hello Silas. I was able to run atlas 2.18.0 however I noticed for some of my samples in `samples.tsv` there are NaNs in the following columns.

`Reads_QC_R1`
`Reads_QC_R2`
`Reads_QC_se`

Not really sure what this means, as the files do exist at corresponding paths for other files. All other columns are filled out appropriately. Writing to see if there is perhaps something I missed and whether this would've had any affect on the downstreams processing. I have QC stats, assemblies, bins and mapping counts for these samples so I'm abit confused. Thanks!

** Atlas version**
2.18.0
**Additional context**

**Comment by SilasK** [comment]
It does not impact the downstream pipline. If the filename is empty (NA) the default path in atlas is used and the pipeline should work.

However, it would still be better to write the correct names there. You said there are only some NA rows. So you know the pattern for imputing them.

I keep this issue open, and try to fix it in a later version.

**Comment by AroArz** [comment]
I'll continue writing here. I'm rerunning atlas, this time with many more samples and atlas is crashing on `qc_reads` complaining about empty values in BinGroup. I've specified about 10 bingroups titled

"BG1", "BG2" ... "BG10", "BGmock"

There are no NaNs in this column and no empty strings. BinGroups are <150 in size. 

![image](https://github.com/metagenome-atlas/atlas/assets/51719360/e9cf43e8-f992-4b30-9824-700da1b17769)
Help appreciated.

**Comment by SilasK** [comment]
I guess where the bug is. 
can you send me the full sample.tsv please?

**Comment by AroArz** [comment]
Sent! When I started the run for the first time, atlas had generated a short string in a new row at the end of the csv which I believe caused the first error, so I promptly removed it. After each `qcread` errror, if I restart the pipeline, the sample which produced the error is resubmitted and completed without errors. Errors however seem to occur for many samples, I’ve restarted it about 10 times so far and it is progressing, slowly.

**Comment by SilasK** [comment]
the step `qcread` essentially, copies the input files to the output files.
And adds the files to the sample.tsv. However multiple threads reading / writing the sample.tsv cause errors. 



I suggest to use this script to move the files yourself.
then delete the .snakemake folder and run `atlas run qc`


```
#!/bin/bash

set -e


# Get a list of all samples with clean_R1 files
samples=$(find . -type d -name "*clean_R1*.fastq.gz" | cut -d/ -f2)

# For each sample, move the input files to the output files
for sample in $samples; do

  for fraction in Rr R2 see; do

    cp -v "$sample/sequence_quality_control/${sample}_clean_${fraction}.fastq.gz" "$sample/sequence_quality_control/${sample}_QC_${fraction}.fastq.gz"
  done

done
```

I will then send you a correct sample.tsv

**Comment by SilasK** [comment]
fixed by atlas v 2.18.1
